### PR TITLE
fix: relation creation API

### PR DESCRIPTION
**Pull request by ishanarya0**

### CODE REVIEW

File: internal/api/v1beta1/relation.go [hunk 1]
```diff
@@ -65,6 +66,13 @@ func (h Handler) CreateRelation(ctx context.Context, request *shieldv1beta1.Crea
 		return nil, grpcBadBodyError
 	}
 
+	if !namespace.IsSystemNamespaceID(request.GetBody().ObjectNamespace) {
```
Comment (mabdh) [inline internal/api/v1beta1/relation.go]: GetBody().GetObjectNamespace() ?

File: internal/api/v1beta1/relation.go [hunk 2]
```diff
@@ -65,6 +66,13 @@ func (h Handler) CreateRelation(ctx context.Context, request *shieldv1beta1.Crea
...
+		_, err := h.resourceService.Get(ctx, request.GetBody().GetObjectId())
```
Comment (mabdh) [inline internal/api/v1beta1/relation.go:70]: What is the purpose of doing this?

should we use the resource id from this get for the flow below?

File: internal/api/v1beta1/relation.go (hunk 1 above)
Comment (ishanarya0) [reply internal/api/v1beta1/relation.go]: sure

File: internal/api/v1beta1/relation.go (hunk 2 above)
Comment (ishanarya0) [reply internal/api/v1beta1/relation.go:70]: Relation creation happens for system-namespace entities and regular resources.

For regular resources
- We want to check if the ObjectId sent (which can be in the format of uuid but still can be a resource name - case of guardian appeal), is a valid resource id or not.

For system namespace
- we don't need any extra check in the exisiting flow

`should we use the resource id from this get for the flow below?`
we are not checking this for system ns, so we can't use it below

**Review by mabdh — APPROVED** [review]

**Pull request merged**